cmd/airflow_test.go

```diff
@@ -224,7 +222,77 @@ func (s *AirflowSuite) mockUserInput(i string) (r, stdin *os.File) {
 }
 
 func (s *AirflowSuite) TestAirflowInit() {
+	TemplateList = func() ([]string, error) { return []string{"A", "B", "C", "D"}, nil }
+	s.Run("initialize template based project via select-template flag", func() {
+		config.WorkingPath = s.tempDir
+		cmd := newAirflowInitCmd()
+		cmd.Flag("name").Value.Set("test-project-name")
+		cmd.Flag("select-template").Value.Set("true")
+		var args []string
+
+		input := []byte("1")
+		r, w, inputErr := os.Pipe()
+		s.Require().NoError(inputErr)
+		_, writeErr := w.Write(input)
+		s.NoError(writeErr)
+		w.Close()
+		stdin := os.Stdin
+		// Restore stdin right after the test.
+		defer func() { os.Stdin = stdin }()
+		os.Stdin = r
+		err := airflowInit(cmd, args)
+		s.NoError(err)
+	})
+
+	s.Run("invalid template name", func() {
+		config.WorkingPath = s.tempDir
+		cmd := newAirflowInitCmd()
+		cmd.Flag("name").Value.Set("test-project-name")
+		cmd.Flag("from-template").Value.Set("E")
+		var args []string
+
+		r, stdin := s.mockUserInput("y")
+
+		// Restore stdin right after the test.
+		defer func() { os.Stdin = stdin }()
+		os.Stdin = r
+		err := airflowInit(cmd, args)
+		s.EqualError(err, "E is not a valid template name. Available templates are: [A B C D]")
+	})
+
+	s.Run("successfully initialize template based project ", func() {
+		airflow.ExtractTemplate = func(templateDir, destDir string) error {
+			err := os.MkdirAll(destDir, os.ModePerm)
+			s.NoError(err)
+			mockFile := filepath.Join(destDir, "requirements.txt")
+			file, err := os.Create(mockFile)
+			s.NoError(err)
+			defer file.Close()
+			_, err = file.WriteString("test requirements file.")
+			s.NoError(err)
+			return nil
+		}
+		config.WorkingPath = s.tempDir
+		cmd := newAirflowInitCmd()
+		cmd.Flag("name").Value.Set("test-project-name")
+		cmd.Flag("from-template").Value.Set("A")
+		var args []string
+
+		r, stdin := s.mockUserInput("y")
+
+		// Restore stdin right after the test.
+		defer func() { os.Stdin = stdin }()
+		os.Stdin = r
+		err := airflowInit(cmd, args)
+		s.NoError(err)
+
+		b, _ := os.ReadFile(filepath.Join(s.tempDir, "requirements.txt"))
+		fileContents := string(b)
+		s.True(strings.Contains(fileContents, "test requirements file"))
+	})
+
 	s.Run("success", func() {
+		config.WorkingPath = s.tempDir
```

Comment:
this could be part of setup, wdyt?

Reply:
Thanks @neel-astro ! I've updated the tests now.